### ISSUE TITLE
Can't play RTSP URL containing user authentication in the URL on Android?

### ISSUE THREAD

**Issue by Wudi-Unity** [RenderHeads/UnityPlugin-AVProVideo#470]
**Describe the bug**
I use Assets/AVProVideo/Prefabs/FullscreenVideo to my game scene.
Source Path set "Absolute Path Or URL" and key the RTSP's URL (rtsp://rtmp.infomaniak.ch/livecast/latele)
At "Preferred Video API", I select "Media Player"
It doesn't show error and build Android success
but this Android APP does not show the streaming movie

How do I set to use this plugin to play RTSP videos?

**Your Setup (please complete the following information):**
 - Unity version: Unity2020.1.6f1
 - AVPro Video version: Version 1.11.4
 - Operating system version: Android
 - Device model: 
 - Video specs (resolution, frame-rate, codec, file size): RTSP

**Comment by AndrewRH** [comment]
Yes, it should be possible to play RTSP on Android using the Media Player API.  

I think that URL is old and is no longer active.

I'm afraid I don't have a link to any other public RTSP feeds right now.

Thanks,

**Comment by Wudi-Unity** [comment]
I use this RTSP url to AVPro plugin
rtsp://admin:ez88888888@59.120.83.220:10012/cam/realmonitor?channel=1&subtype=1

but it shows this error:
[AVProVideo] Error: Loading failed.  File not found, codec not supported, video resolution too high or insufficient system resources

screenshot:
https://drive.google.com/file/d/1zgl-Pu00CNx9BlOOfL2Hh62yj3Sz0mwQ/view?usp=sharing

Can you help me?
thanks~

**Comment by Wudi-Unity** [comment]
I see USER MANUAL say:

Also note that starting with Android 9 (API level 28) cleartext support (unencrypted
connections) is disabled by default, which can cause some HTTP streams to fail. You
should be able to resolve this by switching the URL to HTTPS or by adding
android:usesCleartextTraffic="true" into your AndroidManifest.xml file

so...
maybe I can't use RTSP
must be to use Https?

**Comment by AndrewRH** [comment]
That is just for HTTP - so for HTTP you need to use HTTPS or use that workaround on the newer Androids (same on Apple platform now).

With RTSP it is different.  I believe the problem is caused by having the username + password in the URL.  If you are able to test without any password and remove those fields then I think it will work (when switching to the MediaPlayer API instead of the default ExoPlayer API).

You see the Android RTSP support doesn't include authentication support. See open issues here:
https://issuetracker.google.com/issues/36920753

And here is another discussion about it:
https://stackoverflow.com/questions/42573257/rtsp-stream-in-android-with-username-and-password

So basically, it might work if you remove authentication. 

Otherwise you'll have to use another protocol (not RTSP), or use another player that does support authentication.

Thanks,

**Comment by Wudi-Unity** [comment]
Oh No~
My RTSP URL has username and password><
And I can't remove it.
but...
If  I want to build Android and IOS
Can I use m3u8's URL at this plugin? (or MPEG?)

**Comment by AndrewRH** [comment]
Yes m3u8 URLS are very well supported by the plugin.

MPEG is an old format and isn't supported, unless you mean .mp4 - in which case it would play.

Does that help?

Thanks,

**Comment by Wudi-Unity** [comment]
I use m3u8'URL
In unity simulation , it can play normally
but if I build an Android App, it can't play the streaming movie.
What could be the reason?
Thanks~

**Comment by RenderHeadsSte** [comment]
Can you provide a link to one of your HLS streams?

**Comment by Wudi-Unity** [comment]
My m3u8's URL is : 
http://139.162.82.207:5080/LiveApp/streams/804812567471325976934616.m3u8

And this is my setting:
https://drive.google.com/file/d/1wwcQq2i00VRh_Bfe5mrA0IN7hcATEh_h/view?usp=sharing

https://drive.google.com/file/d/1Cy58zIL6666OZtF4H9KnFICzxkQ7k1Zm/view?usp=sharing

ps.
My Unity version is:
2020.1.6f1

**Comment by RenderHeadsSte** [comment]
Your URL works first time just as it should do in AVPro v1 using the in1_Demo_BackgroundShader scene.  All I did was select to use the MediaPlayer API path (rather than ExoPlayer) and set your URL (see screen shot).  I also tested it in AVPro v2 (latest beta) and it works at it should.

![image](https://user-images.githubusercontent.com/46531149/94546756-3e3bba00-0246-11eb-814b-74f194e8fc84.png)

**Comment by RenderHeadsSte** [comment]
I can report that the same setup (Demo 01, MediPlayer API path, your URL) works in Unity 2020.1.6f1.

**Comment by AndrewRH** [comment]
@Wudi-Unity If you're using a newer version of Android then you will need a HTTPS or to implement the notes in the user manual:
Also note that starting with Android 9 (API level 28) cleartext support (unencrypted
connections) is disabled by default, which can cause some HTTP streams to fail. You
should be able to resolve this by switching the URL to HTTPS or by adding
android:usesCleartextTraffic="true" into your AndroidManifest.xml file

If this isn't the case then we would need to see the android log (abd logcat) as there will be an error message explainign why it can't open the URL.

I hope this information helps.

This discussion has veered away from the original issue about RTSP streaming though so I will close it soon.  If you're having a problem with HLS (m3u8) please open a new issue.

Thanks,

**Comment by Wudi-Unity** [comment]
Thank you for your reply and help.
but I can't use RTSP and m3u8
so...I create a new Unity project
And Import  AVPro plugin to my scene
use "01_Demo_BackgroundShader" and build Android's APP

my unity  simulation screenshot is:
https://drive.google.com/file/d/1FA0EKRgPY9x1-KfjL3ZG1AdN6k6ykuLL/view?usp=sharing

but my Android Phone screenshot is:
https://drive.google.com/file/d/1rXXxdYZL6XupMdjf0L9QbDCAmFJk9Fks/view?usp=sharing

What could be the reason?

thanks~

**Comment by Wudi-Unity** [comment]
> 我可以報告說相同的設置（Demo 01，MediPlayer API路徑，您的URL）在Unity 2020.1.6f1中有效。

but I can't work.

this is my setting:
player setting --> other settings --> Graphics APIs ==> remove "Vulkan"
internet access: select "Require"
Configuration --> Scripting Backend : select "IL2CPP"
Target Architectures: Select "ARMv7" and "ARM64"
Minimum API Level: select "Android 4.4(API level 19)

**Comment by RenderHeadsSte** [comment]
I have requested access to the google drive links you posted, but as yet you have not granted access.

It isn't the fact you are using IL2CPP...because that works fine in tests here.  My hunch is that it is the ARM64 plugin metafile issue.  I could verify this if you posted up a logcat log.

Failing that, try this fix outlined [here](https://github.com/RenderHeads/UnityPlugin-AVProVideo/issues/112#issuecomment-509605543):

**Comment by Wudi-Unity** [comment]
sorry, you can see the screenshot picture now.
And this is my Plugins folder information:
https://drive.google.com/file/d/1aPn88_oqbbs18Fi8p1qgfdrTWOji0S6k/view?usp=sharing

**Comment by Wudi-Unity** [comment]
> adding
android:usesCleartextTraffic="true" into your AndroidManifest.xml file

Is this file in my project?

**Comment by RenderHeadsSte** [comment]
You have to add a manifest (I believe) at location  \Assets\Plugins\Android\AndroidManifest.xml

There are plenty of threads out there already on the topic.  [Here](https://forum.unity.com/threads/android-9-cleartext-http-traffic-to-my_ip-not-permitted.594298/) is one I pulled at random :)

**Comment by RenderHeadsSte** [comment]
> sorry, you can see the screenshot picture now.
> And this is my Plugins folder information:
> https://drive.google.com/file/d/1aPn88_oqbbs18Fi8p1qgfdrTWOji0S6k/view?usp=sharing

You need to ensure that each of the arm64-v8a plugin files in that screenshot are set to ARM64 architecture as per the link I sent.

**Comment by RenderHeadsSte** [comment]
We shall be closing this issue shortly due to inactivity

**Comment by Hettylool** [comment]
Hi，I also have this problem.
my avpro version is 1.11.4
and my rtsp url without username and password
i  play RTSP on Android using the Media Player API
but it seem only work on unity editor ,not work on may mobile phone 
with errror:[AVProVideo] Error: Loading failed. File not found, codec not supported, video resolution too high or insufficient system resources

**Comment by RenderHeadsSte** [comment]
@Hettylool Are you able to send over the URL?

**Comment by Hettylool** [comment]
I built a LAN
URL can only work in LAN
URL:rtsp://10.10.21.171/test2
